src/apphosting/yaml.ts

```diff
@@ -81,53 +77,44 @@ export class AppHostingYamlConfig {
     }
   }
 
-  writeToFile(customFilePath?: string) {
-    if (!this.filePath && !customFilePath) {
-      throw Error("No apphosting yaml file path provided to write to");
+  /**
+   * Loads the given file if it exists and updates it. If
+   * it does not exist a new file will be created.
+   */
+  async writeToFile(filePath: string) {
```

Comment:
naming nit: this feels somewhat surprising. I would rename this to something like `upsertFile`, `write` _usually_ indicates a full overwrite